### PR TITLE
Refactor and cleanup dev guide

### LINKED ISSUE #3264: Clean up developer guide
At the suggestion of @bryevdv, the "Environment Variables" section makes more sense under "Building and Installing", especially for new developers.

I have a separate PR that moves this section and also corrects a few grammatical and spelling errors.

### PR DESCRIPTION
**Pull request by jbcrail**
issues: fixes #3264.

### CODE REVIEW

**Comment by bryevdv** [comment]
We are running very tight on TravisCI resources at the moment. Since this is a docs only PR I have cancelled the automated build. 

Also just FYI please check out the following docs regarding tagging issues and PRs:

https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management

**Comment by bryevdv** [comment]
@jbcrail is this PR ready or still WIP?

**Comment by jbcrail** [comment]
@bryevdv it's ready. Thanks for the FYI. I'll make sure to follow the BEP next time.

**Comment by bryevdv** [comment]
@jbcrail my one suggestion is that now the title "Building and Installing" is a bit of a misnomer. Maybe "Getting Set Up" ? and then rename `building.rst` -> `setup.rst`? Other thoughts welcome.

**Comment by jbcrail** [comment]
@bryevdv the title change sounds good. I made the update.

I noticed the docs still refer to non-existent `BOKEH_DEV` and `bokeh-server`. Should this be updated now or in a separate PR?

**Comment by bryevdv** [comment]
Definitely remove anything related to the old `bokeh-server` @mattpap can you say if `BOKEH_DEV` currently a thing that works or not?

**Comment by mattpap** [comment]
`BOKEH_DEV` does work (well almost, minor bug introduced in server PR). I recently added note about it to the documentation. I'm actually confused with the question, using `BOKEH_DEV` is the way to do development in bokeh (excluding server).

**Comment by bryevdv** [comment]
@mattpap my habit is just to set the env vars I want individually, and I feel like I tried `BOKEH_DEV` recently and it did not work like I expected. But I will try to change my habits. 

@jbcrail OK almost done then I think, can you also add a section for `BOKEH_DEV` to the environment variables section? I know it duplicates a little the "Developing Examples" section but I'd just like to make sure all the env vars are listed in the env vars section even if they are also talked about elsewhere.

File: sphinx/source/docs/dev_guide/setup.rst [hunk 1]
```diff
@@ -296,9 +296,79 @@ invocation:
...
+    standard library webbrowser_ module.
+
+* ``BOKEH_DOCS_CDN`` --- What version of BokehJS to use when building sphinx
+    docs locally.
+
+    .. note::
+        Set to ``"local"`` to use a locally built dev version of BokehJS.
+
+* ``BOKEH_DOCS_VERSION`` --- What version of Bokeh to show when building sphinx
+    docs locally. Useful for re-deployment purposes.
```
Comment (bryevdv) [inline sphinx/source/docs/dev_guide/setup.rst]: Both `BOKEH_DOCS_CDN` and `BOKEH_DOCS_VERSION` are only useful when running `make` to build the docs, would be good to add a note about that I think.

**Comment by jbcrail** [comment]
@mattpap Newbie error on my part. I did a cursory search of the docs and code for `BOKEH_DEV`. I didn't see the flag in `bokeh/settings.py`. For the docs, what env vars are enabled by `BOKEH_DEV`?

**Comment by mattpap** [comment]
@jbcrail, `BOKEH_DEV` is a meta variable. In `bokeh.settings` it lives under `_is_dev` property. All `_get_*` methods allow third parameter `dev`, which is used instead of the default value when `BOKEH_DEV` is set to `true`. You should be able to figure out which settings are modified by `BOKEH_DEV` by now, but in a quick summary: `browser`, `resources`, `minified`, `log_level`,  `pretty` and `simple_ids`.

**Comment by mattpap** [comment]
btw. `pretty` (json pretty printing) is broken by recent changes.

File: sphinx/source/docs/dev_guide/setup.rst (hunk 1 above)
Comment (jbcrail) [inline sphinx/source/docs/dev_guide/setup.rst]: Good suggestions. I added a note to both variables.

**Comment by jbcrail** [comment]
@bryevdv I added the `BOKEH_DEV` section. I also replaced the references to `bokeh-server`.

**Comment by bryevdv** [comment]
@jbcrail great thanks, merging now

**Pull request merged**